src/simsopt/geo/permanent_magnet_grid.py

```diff
@@ -0,0 +1,519 @@
+        if self.coordinate_flag == 'cylindrical':
+            RPhiZ_grid = self._setup_uniform_rz_grid()
+            self.final_grid, self.inds = sopp.define_a_uniform_cylindrical_grid_between_two_toroidal_surfaces(
+                contig(2 * np.pi * self.phi), 
+                contig(normal_inner), 
+                contig(normal_outer),
+                contig(RPhiZ_grid), 
+                contig(self.r_inner), 
+                contig(self.r_outer), 
+                contig(self.z_inner), 
+                contig(self.z_outer)
+            )
+            # define_a_uniform_cylindrical_grid_between_two_toroidal_surfaces only returns the inds to chop at
+            # so need to loop through and chop the grid now
+            self.inds = np.array(self.inds, dtype=int)
+            for i in reversed(range(1, len(self.inds))):
+                for j in range(0, i):
+                    self.inds[i] += self.inds[j]
+            self.ndipoles = self.inds[-1]
+            final_grid = []
+            for i in range(self.final_grid.shape[0]):
+                if not np.allclose(self.final_grid[i, :], 0.0):
+                    final_grid.append(self.final_grid[i, :])
+            self.final_grid = np.array(final_grid)
+            cell_vol = self.final_grid[:, 0] * self.dr * self.dz * 2 * np.pi / (self.nphi * self.plasma_boundary.nfp * 2)
+            # alternatively, weight things roughly by the minor radius
+            # cell_vol = np.sqrt((dipole_grid_r - self.plasma_boundary.get_rc(0, 0)) ** 2 + dipole_grid_z ** 2) * self.dr * self.dz * 2 * np.pi / (self.nphi * self.plasma_boundary.nfp * 2)
+
+            self.m_maxima = B_max * cell_vol / mu0
+            self.dipole_grid_xyz = np.zeros(self.final_grid.shape)
+            self.dipole_grid_xyz[:, 0] = self.final_grid[:, 0] * np.cos(self.final_grid[:, 1])
+            self.dipole_grid_xyz[:, 1] = self.final_grid[:, 0] * np.sin(self.final_grid[:, 1])
+            self.dipole_grid_xyz[:, 2] = self.final_grid[:, 2]
+            self.pm_phi = self.final_grid[:, 1]
+        else:
+            # If some other coordinate_flag, use a retangular Cartesian grid
+            self._setup_uniform_grid()
+            self.dipole_grid_xyz = sopp.define_a_uniform_cartesian_grid_between_two_toroidal_surfaces(
+                contig(normal_inner), 
+                contig(normal_outer), 
+                contig(self.xyz_uniform), 
+                contig(self.xyz_inner), 
+                contig(self.xyz_outer)
+            )
+            inds = np.ravel(np.logical_not(np.all(self.dipole_grid_xyz == 0.0, axis=-1)))
+            self.dipole_grid_xyz = self.dipole_grid_xyz[inds, :]
+            self.ndipoles = self.dipole_grid_xyz.shape[0]
+            self.pm_phi = np.arctan2(self.dipole_grid_xyz[:, 1], self.dipole_grid_xyz[:, 0])
+            cell_vol = self.dx * self.dy * self.dz 
+            self.m_maxima = B_max * cell_vol / mu0 * np.ones(self.ndipoles)
```

Comment:
Please consider promoting m_maxima to an input either in `PermanentMagnetGrid` and/or `geo_setup`

Reply:
See above, will fix this shortly.